rfcs/034-location_location_location/README.md

```diff
@@ -0,0 +1,218 @@
+        "type": "AccessStatus"
+      },
+      "terms": "The papers are available subject to the usual conditions of access to Archives and Manuscripts material.",
+      "type": "AccessCondition"
+    }
+  ],
+  "type": "PhysicalLocation"
+}
+```
+
+A single Miro image (a IIIF image location):
+
+```json
+{
+  "locationType": {
+    "id": "iiif-image",
+    "label": "IIIF Image API",
+    "type": "LocationType"
+  },
+  "url": "https://iiif.wellcomecollection.org/image/A0000681.jpg/info.json",
+  "credit": "Royal Veterinary College",
+  "license": {
+    "id": "cc-by-nc",
+    "label": "Attribution-NonCommercial 4.0 International (CC BY-NC 4.0)",
+    "url": "https://creativecommons.org/licenses/by-nc/4.0/",
+    "type": "License"
+  },
+  "accessConditions": [],
+  "type": "DigitalLocation"
+}
+```
+
+A digitised item (a IIIF Presentation location):
+
+```json
+{
+  "locationType": {
+    "id": "iiif-presentation",
+    "label": "IIIF Presentation API",
+    "type": "LocationType"
+  },
+  "url": "https://wellcomelibrary.org/iiif/b22398752/manifest",
+  "license": {
+    "id": "pdm",
+    "label": "Public Domain Mark",
+    "url": "https://creativecommons.org/share-your-work/public-domain/pdm/",
+    "type": "License"
+  },
+  "accessConditions": [
+    {
+      "status": {
+        "id": "open",
+        "label": "Open",
+        "type": "AccessStatus"
+      },
+      "type": "AccessCondition"
+    }
+  ],
+  "type": "DigitalLocation"
+}
+```
+
+</details>
+
+
+
+## What should a location tell you?
+
+When somebody looks at an item, here's what they want to know about location:
+
+*   **Am I allowed to see it?**
+    This is encoded by the *access conditions* field, which includes an *access status* (drawn from a fixed vocabulary: open, closed, requires-permission, etc.) and an *access description* (free-text, written by cataloguers).
+
+*   **If I'm allowed to see it, where can I find it?**
+    For example: where it sits on the physical shelves within the library, or a particular URL, or a IIIF resource.
+
+*   **If I'm allowed to see it, can I use it?**
+    This is the *license* on the item.
+
+    Note: Our current model only supports a license on digital locations.
+    Is this correct, or do we need to support licenses on physical locations also?
+
+We need to support filtering and aggregating by location.
+For example, users should be able to find all the material that's openly available online (if they're working remotely), or all the material in the library (if they're in the building).
+
+
+
+## What locations do we need to support?
+
+These are the types of location our model needs to support:
+
+*   **Open shelves.**
+
+    This should include a *shelf location* (e.g. History of Medicine Collection) and a *shelfmark* (e.g. TW.AL) so that a user can find it.
+
+*   **Closed stores.**
+
+    Although the MARC records include a shelf location and a shelfmark, the Collections team would prefer this not to be exposed publicly.
```

Comment:
I may have overstated this preference, not helped by the indiscriminate use of 'shelfmark' in Wellcome practice to refer to many different things:

- An item identifier e.g. EBP/A/1234, EPH4 - this _does_ need to be exposed publicly
- A non-classified shelfmark, usually a number in sequence with a prefix for the series e.g. M1234, K1234 - this _could_ be exposed publicly, but I'd argue is meaningless to most users
- A broadly classified shelfmark e.g. AFW.AA1 - as above, no harm in exposing but meaningless (also to staff in some instances...)
- A number within a sequence which is only interpretable in combination with shelf location e.g. 3
- The concatenated combination of two separate fields in Calm e.g. PP/SUL/A/3/4/1:Box25 - this _doesn't_ need to be exposed, and personally I think its confusing to users (and potentially a security risk) to do so

There's also:
- no item shelfmark (all visual material) where equivalent data is in a free-text field _so that_ it can't be exposed publicly

This may not be an exhaustive list. Now may not be the best time to tackle all of this either, so perhaps you should look at it the other way round - expose `949` by default with exceptions where user-relevant information is already available elsewhere in the model?

Reply:
So I think I understand what you're suggesting:

* We have a place to put identifiers, which are usually pulled from other places – although I can't find EPB/A/1234 or EPH4 on wl.org or wc.org. Are those made-up identifiers, or has something got lost?
* The other four instances of "shelfmark" are things that are either meaningless or dangerous to expose on closed shelves, so either way let's not.

What's in MARC 949? OCLC says it's reserved for local use – what do Wellcome use it for?

Reply:
It is indeed a shelfmark. The problem is more around how inconsistently it has been used. 

EPH4 is https://search.wellcomelibrary.org/iii/encore/record/C__Rb1538878?lang=eng
EPB/A/1234 I made up and is typically therefore missing from the sequence, but here's EPB/A/1235 https://search.wellcomelibrary.org/iii/encore/record/C__Rb2139241?lang=eng

Reply:
We already present these "shelfmarks" on wellcomelibrary.org, e.g. https://catalogue.wellcomelibrary.org/record=b1818611

As such, I don't think there's any additional harm in exposing them for now. I've updated the RFC to say:

> For now, this will be populated on all "PhysicalLocation" records, including ClosedStores. We may choose to remove it from some locations later, after appropriate discussions with Collections Information to API still includes all the identifiers and information required by users.